aws-opentelemetry-distro/src/amazon/opentelemetry/distro/exporter/aws/metrics/aws_cloudwatch_emf_exporter.py

```diff
@@ -0,0 +1,523 @@
+        if unit in self.EMF_SUPPORTED_UNITS:
+            return unit
+
+        # Map from OTel unit to CloudWatch unit
+        mapped_unit = self.UNIT_MAPPING.get(unit)
+
+        return mapped_unit
+
+    def _get_dimension_names(self, attributes: Attributes) -> List[str]:
+        """Extract dimension names from attributes."""
+        # Implement dimension selection logic
+        # For now, use all attributes as dimensions
+        return list(attributes.keys())
+
+    def _get_attributes_key(self, attributes: Attributes) -> str:
+        """
+        Create a hashable key from attributes for grouping metrics.
+
+        Args:
+            attributes: The attributes dictionary
+
+        Returns:
+            A string representation of sorted attributes key-value pairs
+        """
+        # Sort the attributes to ensure consistent keys
+        sorted_attrs = sorted(attributes.items())
+        # Create a string representation of the attributes
+        return str(sorted_attrs)
+
+    def _normalize_timestamp(self, timestamp_ns: int) -> int:
+        """
+        Normalize a nanosecond timestamp to milliseconds for CloudWatch.
+
+        Args:
+            timestamp_ns: Timestamp in nanoseconds
+
+        Returns:
+            Timestamp in milliseconds
+        """
+        # Convert from nanoseconds to milliseconds
+        return timestamp_ns // 1_000_000
+
+    def _create_metric_record(self, metric_name: str, metric_unit: str, metric_description: str) -> MetricRecord:
+        """
+        Creates the intermediate metric data structure that standardizes different otel metric representation
+        and will be used to generate EMF events. The base record
+        establishes the instrument schema (name/unit/description) that will be populated
+        with dimensions, timestamps, and values during metric processing.
+
+        Args:
+            metric_name: Name of the metric
+            metric_unit: Unit of the metric
+            metric_description: Description of the metric
+
+        Returns:
+            A MetricRecord object
+        """
+        return MetricRecord(metric_name, metric_unit, metric_description)
+
+    def _convert_gauge(self, metric: Metric, data_point: NumberDataPoint) -> MetricRecord:
+        """Convert a Gauge metric datapoint to a metric record.
+
+        Args:
+            metric: The metric object
+            data_point: The datapoint to convert
+
+        Returns:
+            MetricRecord with populated timestamp, attributes, and value
+        """
+        # Create base record
+        record = self._create_metric_record(metric.name, metric.unit, metric.description)
+
+        # Set timestamp
+        try:
+            timestamp_ms = (
+                self._normalize_timestamp(data_point.time_unix_nano)
+                if data_point.time_unix_nano is not None
+                else int(time.time() * 1000)
+            )
+        except AttributeError:
```

Comment:
non-blocking - Just wondering if these try-catches are overkill, as the `NumberDataPoint` [should have these attributes](https://github.com/open-telemetry/opentelemetry-python/blob/59baf3102ab99e3234fd136abb157836a383ec9f/opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/point.py#L30-L42).